controllers/healthcheck_controller.go

```diff
@@ -90,23 +90,50 @@ func (r *HealthCheckReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error)
 	ctx := context.Background()
 	log := r.Log.WithValues(hcKind, req.NamespacedName)
 
+	log.Info("Starting HealthCheck reconcile ...", "for this", "hc")
+
 	// initialize timers map if not already done
 	if r.RepeatTimersByName == nil {
 		r.RepeatTimersByName = make(map[string]*time.Timer)
 	}
 
-	var healthCheck activemonitorv1alpha1.HealthCheck
-	if err := r.Get(ctx, req.NamespacedName, &healthCheck); err != nil {
+	var healthCheck = &activemonitorv1alpha1.HealthCheck{}
+	if err := r.Get(ctx, req.NamespacedName, healthCheck); err != nil {
 		// if our healthcheck was deleted, this Reconcile method is invoked with an empty resource cache
 		// see: https://book.kubebuilder.io/cronjob-tutorial/controller-implementation.html#1-load-the-cronjob-by-name
-		log.Info("Healthcheck object not found for reconciliation, likely deleted")
+		log.Info("Healthcheck object not found for reconciliation, likely deleted", "deleted", "deleted")
 		// stop timer corresponding to next schedule run of workflow since parent healthcheck no longer exists
 		if r.RepeatTimersByName[req.NamespacedName.Name] != nil {
-			log.Info("Cancelling rescheduled workflow for this healthcheck due to deletion")
+			log.Info("Cancelling rescheduled workflow for this healthcheck due to deletion", "deleted", "deleted")
 			r.RepeatTimersByName[req.NamespacedName.Name].Stop()
 		}
 		return ctrl.Result{}, ignoreNotFound(err)
 	}
+
+	return r.execHealthCheck(ctx, req, log, healthCheck)
+}
+
+func (r *HealthCheckReconciler) execHealthCheck(ctx context.Context, req ctrl.Request, log logr.Logger, healthCheck *activemonitorv1alpha1.HealthCheck) (ctrl.Result, error) {
```

Comment:
looks like this branch may yet first need to pull in changes from other PR (now merged to `master`)

Reply:
yes.. Merged it and resolved conflicts.. should be good now